src/matrix/dense_faer_serial.rs

```diff
@@ -0,0 +1,139 @@
+use std::ops::{Mul, MulAssign};
+
+use super::{DenseMatrix, Matrix, MatrixCommon, MatrixView, MatrixViewMut};
+use crate::scalar::{IndexType, Scalar, Scale};
+use anyhow::Result;
+use faer::{linalg::matmul::matmul, Col, ColMut, ColRef, Mat, MatMut, MatRef, Parallelism};
+
+macro_rules! impl_matrix_common {
+    ($mat_type:ty) => {
+        impl<'a, T: Scalar> MatrixCommon for $mat_type {
+            type T = T;
+            type V = Col<T>;
+
+            fn ncols(&self) -> IndexType {
+                self.ncols()
+            }
+
+            fn nrows(&self) -> IndexType {
+                self.nrows()
+            }
+        }
+    };
+}
+
+impl_matrix_common!(MatMut<'a, T>);
+impl_matrix_common!(MatRef<'a, T>);
+impl_matrix_common!(Mat<T>);
+
+macro_rules! impl_mul_scale {
+    ($mat_type:ty) => {
+        impl<'a, T: Scalar> Mul<Scale<T>> for $mat_type {
+            type Output = Mat<T>;
+
+            fn mul(self, rhs: Scale<T>) -> Self::Output {
+                self * faer::scale(rhs.value())
```

Comment:
I think it would be neater to impl `From<faer::Scale>` for `Scale<T>` and then have `self * rhs.into()`

Reply:
There are multiple implementations of into for that type, the compiler is requesting further type annotation. The best I could do is something like this `self * Into::<faer::Scale<T>>::into(rhs)`

Reply:
The function will look like

```rust
fn mul(self, rhs: Scale<T>) -> Self::Output {
        self * Into::<faer::Scale<T>>::into(rhs)
 }
```

A different approach that would also work is:

```rust
fn mul(self, rhs: Scale<T>) -> Self::Output {
        let scale: faer::Scale<T> = rhs.into();
        self * scale
}
```

Reply:
omg, that isn't neater at all! I don't see why it can't infer the type you are converting to, but I guess it doesn't :( Its an extra line, but lets go with the second approach, it would still be valuable to keep the conversion impl of Scale -> fear::Scale in only one palce